### PR TITLE
OpenID Connect support

### LINKED ISSUE #473: OpenID Connect support
As a user of the SDP I want to be able to hook up Trino with my exsiting Keycloak or other OIDC comaptible identity provider.

[OAuth 2.0 - Trino docs](https://trino.io/docs/current/security/oauth2.html)

Config we got so far (for keycloak):

        # web-ui.authentication.type: oauth2
        http-server.authentication.type: oauth2
        http-server.authentication.oauth2.client-id: trino
        http-server.authentication.oauth2.client-secret: ${ENV:TRINO_CLIENT_SECRET}
        http-server.authentication.oauth2.issuer: http://${ENV:KEYCLOAK_ADDRESS}/realms/master
        http-server.authentication.oauth2.principal-field: preferred_username

**The relevant branch to start from**: https://github.com/stackabletech/trino-operator/tree/feature/support-oauth2

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Add OpenID Connect support

Closes #473 

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases ~added~ updated
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by fhennig** [comment]
related? https://github.com/stackabletech/trino-operator/issues/473

File: Cargo.toml [hunk 1]
```diff
@@ -29,5 +29,5 @@ strum = { version = "0.25", features = ["derive"] }
 tokio = { version = "1.29", features = ["full"] }
 tracing = "0.1"
 
-# [patch."https://github.com/stackabletech/operator-rs.git"]
-# stackable-operator = { git = "https://github.com/stackabletech//operator-rs.git", branch = "main" }
+[patch."https://github.com/stackabletech/operator-rs.git"]
```
Comment (sbernauer) [inline Cargo.toml:32]: Check that Authclasses are watched!

**Comment by siegfriedweber** [comment]
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/trino-operator-it-custom/114/

**Review by adwk67 — APPROVED** [review]
LGTM! Just one tiny nit.

**Comment by lfrancke** [comment]
Could you please link the documentation here?

**Comment by siegfriedweber** [comment]
https://docs.stackable.tech/home/nightly/trino/usage-guide/security#_oauth2

**Pull request merged**